source/_posts/2024-02-21-voice-chapter-6.markdown

```diff
@@ -0,0 +1,135 @@
+
+Doing wake word detection in HA allows tiny devices like the [M5 ATOM Echo Development Kit][m5-tutorial] to simply stream audio and let all of the processing happen elsewhere.
+The downside is that adding more voice assistants requires more CPU usage in HA as well as more network traffic.
+
+Enter microWakeWord. After listening to an interview with Paulus Schoutsen (founder of Home Assistant) on the [Self Hosted](https://selfhosted.show/) podcast, Kevin Ahrendt created a model based on [Google's Inception neural network](https://towardsdatascience.com/a-simple-guide-to-the-versions-of-the-inception-network-7fc52b863202). As an existing contributor to [ESPHome], Kevin was able to get this new model running on the ESP32-S3 chip inside the S3-BOX-3!
+
+Kevin has trained [three models](https://github.com/esphome/micro-wake-word-models/tree/main/models) for the launch of microWakeWord:
+
+* "okay nabu"
+* "hey jarvis"
+* "alexa"
+
+You can try these out yourself now by following the [ESP32-S3-BOX tutorial][s3-box-tutorial]. Changing the default "okay nabu" wake word will require adjusting your [ESPHome configuration](https://beta.esphome.io/components/micro_wake_word.html) and recompiling the firmware, which may take a long time and requires a machine with more than 2GB of RAM.
+
+We're grateful to Kevin for developing microWakeWord, and making it a part of the open home!
+
+## Sentence trigger responses
+
+Adding custom sentences to Assist is as easy as adding a [sentence trigger][sentence triggers] to an automation. This allows you to trigger any action in Home Assistant with whatever sentences you want.
+
+Now with the new [conversation response] action in HA 2024.2, you can also customize the response spoken or printed back to you. Using [templating](/docs/automation/templating/#sentence), your response can refer to the current state of your home.
+
+<p><img src='/images/blog/2024-02-21-voice-chapter-6/assist-custom-response-editor.png' class='no-shadow' /></p>
+
+You can also refer to [wildcards](/docs/automation/trigger/#sentence-wildcards) in your sentence trigger. For example, the sentence trigger:
+
+```
+play {album} by {artist}
+```
+
+could have the response:
+
+{% raw %}
+```
+Playing {{ trigger.slots.album }} by {{ trigger.slots.artist }}
+```
+{% endraw %}
+
+in addition to calling a media service.
+
+## Improved errors and debugging
+
+Assist users know the phrase "Sorry, I couldn't understand that" all too well. This generic error message was given for a variety of reasons, such as:
+
+* The sentence didn't match any known [intent](https://github.com/home-assistant/intents)
+* The device/area names didn't match
+* There weren't any devices of a specific type in an area (lights, windows, etc.)
+
+Starting in HA 2024.2, Assist provides different error messages for each of these cases.
+
+<p><img src='/images/blog/2024-02-21-voice-chapter-6/error_messages.png' class='no-shadow' /></p>
```

Comment:
```suggestion
<img class="no-shadow" src='/images/blog/2024-02/assist-errors.png' alt='Screenshot showing the new errors Assist will return in case the intention is understood, but something else is missing.'>
```

Reply:
Reusing the same image we put in the last release note that can be found here:
https://www.home-assistant.io/blog/2024/02/07/release-20242/#improved-error-responses-when-using-assist